lib/presentation/pages/login/widgets/custom_form_field.dart

```diff
@@ -0,0 +1,143 @@
+          ],
+          keyboardType: type == FieldType.eMail
+              ? TextInputType.emailAddress
+              : type == FieldType.password
+                  ? TextInputType.name
+                  : type == FieldType.phoneNumber
+                      ? TextInputType.phone
+                      : TextInputType.name,
+          //autovalidateMode: AutovalidateMode.onUserInteraction,
+          obscureText: obscureText ?? false,
+          decoration: InputDecoration(
+              fillColor: Colors.white,
+              filled: true,
+              prefixIcon: Icon(
+                icon,
+                color: Colors.deepPurple,
+              ),
+              suffixIcon: type == FieldType.password
+                  ? IconButton(
+                      icon: Icon(postFixIcon ?? Icons.visibility_off),
+                      onPressed: toggleVisibility,
+                      color: Colors.black,
+                    )
+                  : null,
+              hintText: hintText,
+              labelText: hintText,
+              labelStyle: const TextStyle(color: Colors.deepPurple),
+              floatingLabelBehavior: FloatingLabelBehavior.always,
+              errorStyle: const TextStyle(color: Colors.blue),
+              enabledBorder: const OutlineInputBorder(
+                  borderSide: BorderSide(color: Colors.deepPurple)),
+              disabledBorder: const OutlineInputBorder(
+                  borderSide: BorderSide(color: Colors.deepPurple)),
+              focusedErrorBorder: const OutlineInputBorder(
+                  borderSide: BorderSide(color: Colors.deepPurple)),
+              errorBorder: const OutlineInputBorder(
+                  borderSide: BorderSide(
+                      style: BorderStyle.solid, color: Colors.deepPurple)),
+              contentPadding:
+                  EdgeInsets.only(left: 0, top: 0, bottom: height * 0.02),
+              border: const OutlineInputBorder(
+                  borderRadius: BorderRadius.all(Radius.circular(4)))),
+          validator: validator ??
+              (type == FieldType.eMail
+                  ? (value) {
+                      if (value!.isEmpty) {
+                        return "$hintText Cant be empty";
+                      } else if (value.length < 4) {
+                        return "$hintText Must be greater than 4 charactors long";
+                      } else if (!value.isValidEmail &&
+                          !RegExp(r'(^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$)')
+                              .hasMatch(value)) {
+                        return "Invalid Email/Phone Number";
+                      } else {
+                        return null;
+                      }
+                    }
+                  : type == FieldType.password
+                      ? (value) {
+                          if (value == '' || value == null) {
+                            return "$hintText Cant be empty";
```

Comment:
Why 2 else if used can be made in one also

Reply:
changed 1 unwanted elseif check